### PR TITLE
Task/ccls 1759 consolidate pipelines

### PR DESCRIPTION
**Pull request by PhilDigitalJustice**

### CODE REVIEW

File: .github/workflows/build-main.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+# documentation.
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-gradle
+
+name: CI caab-data-api
+
+on:
+  push:
+    branches-ignore:
+      - main
```
Comment (keith-h) [inline .github/workflows/build-main.yml:13]: if this is `build-main` we shouldn't ignore main?

File: .github/workflows/build-main.yml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            aws-region: ${{ vars.DEV_ECR_REGION }}
+
+        - name: Login ecr
+          uses: aws-actions/amazon-ecr-login@v1
+          id: login-ecr
+
+        # Build and push docker image to container repo
+        - name: Build and push image
+          run: |
+            docker build -t $REGISTRY/$REPOSITORY:$IMAGE_TAG .
```
Comment (keith-h) [inline .github/workflows/build-main.yml:106]: should be ${{. env.IMAGE_TAG }}

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -75,48 +88,55 @@ jobs:
...
             REGISTRY: ${{ steps.login-ecr.outputs.registry }}
-            REPOSITORY: ${{ vars.ECR_REPOSITORY }}
-            IMAGE_TAG: ${{ github.sha }}
+            REPOSITORY: ${{ vars.DEV_ECR_REPOSITORY }}
+            IMAGE_TAG: ${{ needs.define-image-tag.outputs.tag_name }}
+
         - name: delete artifact
           uses: geekyeggo/delete-artifact@v2
           with:
             name: data-api-jar
```
Comment (keith-h) [inline .github/workflows/build.yml:116]: will need a separate job for this when not on main to delete the artifact.  Either that, or we separate it out

Comment (PhilDigitalJustice) [reply .github/workflows/build.yml:116]: resolved

**Review by keith-h — APPROVED** [review]
LGTM

**Pull request merged**